### PR TITLE
Update: Show template sources on templates Dataviews sidebar.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR adds a set of Dynamically generated views to templates dataviews that allows the user to filter the templates by source from the sidebar.

## Screenshot
Screenshot 2024-01-23 at 12 58 22

### CODE REVIEW

File: packages/edit-site/src/components/sidebar/index.js [hunk 1]
```diff
@@ -81,7 +82,10 @@ function SidebarScreens() {
 			<SidebarScreenWrapper path="/patterns">
 				<SidebarNavigationScreenPatterns />
 			</SidebarScreenWrapper>
-			<SidebarScreenWrapper path="/:postType(wp_template|wp_template_part)/all">
+			<SidebarScreenWrapper path="/:postType(wp_template)/all">
```
Comment (oandregal) [inline packages/edit-site/src/components/sidebar/index.js]: The PR that switches templates parts implementation to dataviews has landed. Can this be updated to work for both?

File: packages/edit-site/src/components/page-templates/index.js [hunk 2]
```diff
@@ -155,18 +155,45 @@ function TemplatePreview( { content, viewType } ) {
...
+							{
+								field: 'author',
+								operator: 'in',
+								value: activeView,
+							},
+					  ]
+					: [],
+		} ) );
+	}, [ activeView ] );
+	console.error( { view } );
```
Comment (oandregal) [inline packages/edit-site/src/components/page-templates/index.js:195]:
```suggestion
```

**Comment by oandregal** [comment]
This is now working for templates & parts:

https://github.com/WordPress/gutenberg/assets/583546/01381a36-a24d-4231-b5e8-4937abf846cc

**Comment by oandregal** [comment]
There's one thing missing: when clicking one of the sidebar items, the `layout` URL parameter is cleared. This is problematic, specially in the list view:

- go to list view
- click one of the sidebar items
- observe how the preview disappears

**Review by oandregal — APPROVED** [review]

**Comment by jorgefilipecosta** [comment]
Thank you a lot for the review and enhancements @oandregal 😊

**Comment by jameskoster** [comment]
Thanks for working on this @jorgefilipecosta, it certainly gets us a step closer to closing https://github.com/WordPress/gutenberg/issues/58111. 

There are a couple of details missing:

1. For the 'All templates' view could we use the `file` icon to differentiate it from the theme views?
2. It would be nice to include the counts.
3. Perhaps most important: ideally all user-generated templates would be grouped together in a view called "Custom", rather than individual views per user. This view would be the equivalent to filtering by author and selecting all users.

What do you think?

File: packages/edit-site/src/components/sidebar-dataviews/dataview-item.js [hunk 3]
```diff
@@ -29,14 +29,15 @@ export default function DataViewItem( {
...
-		params: { path },
+		params: { path, layout },
 	} = useLocation();
 
 	const iconToUse =
 		icon || VIEW_LAYOUTS.find( ( v ) => v.type === type ).icon;
 
 	const linkInfo = useLink( {
 		path,
+		layout,
```
Comment (ntsekouras) [inline packages/edit-site/src/components/sidebar-dataviews/dataview-item.js:40]: Why do we need this change? In pages list it certainly creates problems(at least in pages list) when changing views. For example if we change a view from the sidebar which has a `grid` view set, I'd expect this to be respected without taking into account the current `layout` param set.

In general the logic is hard to follow with side effects, when handling the views in sidebar and it would be great if we can simplify and also have more consistency between the different lists(pages, templates).
